### ISSUE TITLE
[Bug?] Predictive Variances Are Large At Training Points With FixedNoiseGaussianLikelihood, Small Noise

### ISSUE THREAD

**Issue by CarloGraziani** [cornellius-gp/gpytorch#1918]
# 🐛 Bug
Hi.

Possibly I'm misunderstanding the code usage. I'm attempting use a very simple 1-d case with a few training points and heteroskedastic noise, to verify the limiting case, as the noise becomes small, where the predictive variance should vanish as the training points are approached.  Instead I'm seeing that the variance "snake" still bulges out by a  considerable amount at the training points.  This seems very obviously a chair-to-keyboard-interface issue, but I'm going crazy trying to find the problem, so please excuse me resorting to a "bug" report.

## To reproduce

** Code snippet to reproduce **
```python
# Your code goes here
# Please make sure it does not require any external dependencies (other than PyTorch!)
# (We much prefer small snippets rather than links to existing libraries!)
import torch
import gpytorch
import numpy as np
from matplotlib import pyplot as plt
from numpy.random import default_rng

rng = default_rng(seed=125245)
training_iter = 5000000
lr = 1.0E-4
target = 0.65
tolerance = 0.025
noise = 0.008
test_x = torch.tensor(np.linspace(0.0, 1.0, 50))
x_samp = np.array([])
y_samp = np.array([])

###################################################
def response(x):

    y = 0.8 - 0.45*x + 0.1*np.sin(4.0*np.pi*x)

    return y

y = response(x)

###################################################
def add_training_points(xpts):

    global x_samp, y_samp, train_x, train_y

    xsmp = np.array(xpts)
    ysmp = response(xsmp) + noise * rng.standard_normal(len(xsmp))
    x_samp = np.concatenate((x_samp, xsmp))
    y_samp = np.concatenate((y_samp, ysmp))
    train_x = torch.tensor(x_samp)
    train_y = torch.tensor(y_samp)

###################################################
class ExactGPModel(gpytorch.models.ExactGP):
    def __init__(self, train_x, train_y, likelihood):
        super(ExactGPModel, self).__init__(train_x, train_y, likelihood)
        self.mean_module = gpytorch.means.ConstantMean()
        self.covar_module = gpytorch.kernels.ScaleKernel(gpytorch.kernels.RBFKernel())

    def forward(self, x):
        mean_x = self.mean_module(x)
        covar_x = self.covar_module(x)
        return gpytorch.distributions.MultivariateNormal(mean_x, covar_x)

###################################################
def init_model():
    global likelihood, model, mll, optimizer
    like_noise = torch.ones_like(train_y) * noise
    likelihood = gpytorch.likelihoods.FixedNoiseGaussianLikelihood(noise = like_noise)
    model = ExactGPModel(train_x, train_y, likelihood)
    mll = gpytorch.mlls.ExactMarginalLogLikelihood(likelihood, model)
    #optimizer = torch.optim.Adam(model.parameters(), lr=lr) 
    optimizer = torch.optim.LBFGS(model.parameters(), lr=lr, max_iter=200, history_size=200)


###################################################
def closure():

    with gpytorch.settings.fast_computations(covar_root_decomposition=False, log_prob=False, solves=False):
        optimizer.zero_grad()
        output = model(train_x)
        loss = -mll(output, train_y)
        loss.backward()
    return loss

###################################################
def opt(p0):
    model.train()
    likelihood.train()

    
    model.mean_module.initialize(constant=train_y.mean())
    model.covar_module.initialize(outputscale = p0[0])
    model.covar_module.base_kernel.initialize(lengthscale = p0[1])

    for _ in range(training_iter):
        loss = optimizer.step(closure)

        return loss


###################################################
def print_model_pars():

    print("Output Scale = %f" % model.covar_module.outputscale)
    print("Length Scale = %f" % model.covar_module.base_kernel.lengthscale)
    print("Mean = %f" % model.mean_module.constant)

###################################################
def pred(test_x):

    global lower, upper, mean_fn, lower_y, upper_y, mean_y
    model.eval()
    likelihood.eval()

    with torch.no_grad(), gpytorch.settings.fast_computations(covar_root_decomposition=False, solves=False):
        f_preds = model(test_x)

    mean_fn = f_preds.mean
    lower = (mean_fn - f_preds.stddev).numpy()
    upper = (mean_fn + f_preds.stddev).numpy()
    mean_fn = mean_fn.numpy()

    y_preds = likelihood(f_preds, noise=noise*torch.ones_like(test_x, dtype=torch.float64))
    mean_y = y_preds.mean
    lower_y = (mean_y - y_preds.stddev).numpy()
    upper_y = (mean_y + y_preds.stddev).numpy()
    mean_y = mean_y.numpy()



###################################################
def plotit():
    fig = plt.figure()
    fig.set_figwidth(8.0)
    fig.set_figheight(8.0)

    ax = fig.add_subplot(1,1,1)
    ax.set_xlim(0.0,1.0)
    ax.set_ylim(0.0,1.0)
    rf, = ax.plot(x, y, 'b-', label="Response Function")
    sp = ax.errorbar(x_samp, y_samp, noise, marker='o', capsize=4.0, ls='none', label="Sample Points")

    # ttr = ax.axhspan(target-tolerance, target+tolerance, color='g', alpha=0.5, label="TTR")

    ub = ax.fill_between(test_x.numpy(), lower, upper, color='c', alpha=0.5, label="Uncertainty Box")
    um, = ax.plot(test_x.numpy(), mean_fn, 'r:', alpha=0.5, label="Uncertainty Mean")

    ax.tick_params(axis='x', bottom=False, labelbottom=False)
    ax.tick_params(axis='y', left=False, labelleft=False)
    ax.set_xlabel("Control Space", size='x-large')
    ax.set_ylabel("Design Space", size='x-large')

    ax.legend(handles=[rf, ttr,sp, ub, um])

#############################
x_s = [0.05, 0.21, 0.57, 0.67, 0.8 ]
add_training_points(x_s)

init_model()
p0 = np.array([0.1, 0.05])
loss = opt(p0)
print ("Loss = %f" % loss)
print_model_pars()

pred(test_x)

plotit()
```

** Stack trace/error message **
```
// Paste the bad output here!
The "bad" behavior is in the plot that results. I attach a picture below.
![output](https://user-images.githubusercontent.com/11160521/154355979-b9ab5a9f-4873-47a4-940d-b7c91c9ccdca.png)


```

## Expected Behavior

I actually expected the waist of the variance "snake" to shrink to nearly zero near the training points. I've been looking around for sources of extra variance that might be guarding against instability (jitter/nugget-type things, or noise constraints, etc.) but I can't find what's doing this. The mean is mostly hitting the points, except at the left and right edges of the function (but it should be hitting all of them, nearly head-on, really).

I'm at a loss.  This is the simplest GP problem I've ever put together. What am I doing wrong?

## System information

**Please complete the following information:**
- GPyTorch Version 1.5.0
- PyTorch Version 1.9.0+cpu
- Linux Ubuntu 20.04

**Comment by CarloGraziani** [comment]
Sorry, this was a notebook on my computer, but I couldn't figure out how to get it to show up as a notebook here.

**Comment by CarloGraziani** [comment]
You are absolutely correct. As I said, chair-to-keyboard-interface issue. Noise means variance in that likelihood interface, not standard deviation. I don't know why I was looking  for something deeper. Thanks, "bug" closed.
